pkg/apis/storage/validation/validation.go

```diff
@@ -45,6 +48,10 @@ func ValidateStorageClassUpdate(storageClass, oldStorageClass *storage.StorageCl
 	if storageClass.Provisioner != oldStorageClass.Provisioner {
 		allErrs = append(allErrs, field.Forbidden(field.NewPath("provisioner"), "updates to provisioner are forbidden."))
 	}
+
+	if !reflect.DeepEqual(storageClass.ReclaimPolicy, oldStorageClass.ReclaimPolicy) {
```

Comment:
yes, given that neither can be nil. I will remove this plus the other paranoid checks :)